spinalcordtoolbox/utils/fs.py

```diff
@@ -283,3 +283,4 @@ def mutex(name):
         yield semaphore
     finally:
         semaphore.release()
+        os.remove(semaphore.get_filenames()[0])  # nb: BoundedSemaphore doesn't clean up after itself
```

Comment:
I think this interferes with the functioning of the lock. As a small test, I did this:
1. In one python instance, acquire a `BoundedSemaphore`. There was no wait.
2. In a second python instance, acquire a `BoundedSemaphore` with the same name. There was a wait, then a timeout. So, this scenario works.

In contrast, deleting the file broke the synchronization:
1. In one python instance, acquire a `BoundedSemaphore`. There was no wait.
2. Manually delete the file.
3. In a second python instance, acquire a `BoundedSemaphore` with the same name. There was no wait, but there should have been.

It's unfortunate to be leaving cruft around, but I think it's necessary in this case. At least for unix-likes, I think `/tmp` gets cleared at every reboot, usually. (I have no idea for Windows.)

Reply:
So, I think `portalocker`'s file locks are associated with an inode rather than a filename (at least on Unix-likes, who knows about Windows). To test this, I did:
1. In process A, acquire a `portalocker.Lock` on `somefile`.
2. In process B, try to acquire a lock on `somefile`: this blocks and times out, as expected.
3. Manually, I moved (renamed) the file from `somefile` to `otherfile`. This keeps the same inode, but now a different path refers to it.
4. In process B, try to acquire the lock on `somefile`: this immediately re-creates `somefile` and acquires the lock (as expected, because even though it's the same file name, it's a new inode).
5. In process B, try to acquire a lock on `otherfile`: this blocks and times out, because process A still holds the lock on that inode, from when it was called `somefile`.

So, for `portalocker.BoundedSemaphore` to work properly, we can't delete the associated tmp files, because that breaks the link between the file name (which we use to acquire the lock) and its inode (which is the thing that's actually locked).